S09.03-Exercise-ContentProviderDelete/app/src/androidTest/java/com/example/android/sunshine/SunshineTestCases.java

```diff
@@ -0,0 +1,159 @@
+
+package com.example.android.sunshine;
+
+import android.support.test.espresso.action.ViewActions;
+import android.support.test.espresso.contrib.RecyclerViewActions;
+import android.support.test.espresso.matcher.ViewMatchers;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.widget.ImageButton;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.ViewMatchers.isClickable;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withContentDescription;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.core.AllOf.allOf;
+import static org.hamcrest.core.StringContains.containsString;
+//import static android.despresso_samples.recyclerview.TestUtils.withRecyclerView;
+
+@RunWith(AndroidJUnit4.class)
+
+public class SunshineTestCases {
+
+
+    private String navigateDate = "Tomorrow";
+    private String settings = "Settings";
+    private String title = "Sunshine";
+    private String detail = "Details";
+    private String options = "More options";
+    private String location = "Location";
+    private String temp_Units = "Temperature Units";
+    private String cancel_Button = "Cancel";
+    private String metric_Units = "Metric";
+    private String imperial_Units = "Imperial";
```

Comment:
This would have been a good opportunity to practice creating a model. All of these should be in a model or could even be broken out into separate models such as DetailActivityModel, ActionBarModel etc.

Reply:
All of these string should be called from the resource strings in strings.xml, and as per coding standards these would be within a model. Example for "Settings":
private Clickable<NameOfModel> actionBarSettings = new Clickable<>(NameOfModel.class, withText(R.string.action_settings));